service/share/discovery.go

```diff
@@ -90,7 +90,7 @@ func (d *discovery) findPeers(ctx context.Context) {
 			return
 		case <-t.C:
 			// return if limit was reached to stop the loop and finish discovery
-			if uint(d.set.Size()) >= d.peersLimit {
+			if uint(d.set.Size()) == d.peersLimit {
```

Comment:
It would be nice though to make sure that set.Size never actually exceeds peersLimit bc that would be strange if it did.

Reply:
It won't. TryAdd will not allow to exceed the limit